sound/soc/sof/topology.c

```diff
@@ -3046,8 +3049,26 @@ static void sof_complete(struct snd_soc_component *scomp)
 static int sof_manifest(struct snd_soc_component *scomp, int index,
 			struct snd_soc_tplg_manifest *man)
 {
-	/* not currently parsed */
-	return 0;
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
+
+	/* backward compatible with tplg without ABI info */
+	if (!man->priv.size)
```

Comment:
@bardliao @plbossart Basically, this means I can remove the ABI info from any tplg to clear this hurdle. Is this acceptable?

Reply:
@plbossart yes, exactly. adding a warning is a good idea.